app/components/SCForm.js

```diff
@@ -94,28 +105,64 @@ class SCForm extends Component {
   }
 
   onChange(value) {
-    this.setState({ value });
+    const formInfo = this.props.navigation.state.params.form;
+    var max,
+      min;
+    var length = formInfo.fields.length;
+    // gets the value of what has been entered and expected type
+    for (var i = 0; i < length; i++) {
+      // input
+      field_key = value[formInfo.fields[i].field_key];
+      field_name = formInfo.fields[i].field_key;
+      field_label = formInfo.fields[i].field_label;
+      // expected type
+      type = formInfo.fields[i].type;
+      // max and min values. i.e. length of string or max # of occurences
+      max = formInfo.fields[i].constraints.maximum;
+      min = formInfo.fields[i].constraints.minimum;
+      // if the field_key is expected to be a number, convert it to a number.
+      if (type === 'number') {
+        field_key = +[field_key];
+      }
+    }
+  }
+
+  handleFieldChanged(field_name) {
+    return (e) => {
+      // update() is provided by React Immutability Helpers
+      // https://facebook.github.io/react/docs/update.html
+      let newState = update(this.state, {
+        [field_key]: {
+          $set: e.target.value,
+        },
+      });
+      newState.validationErrors = run(newState, fieldValidations);
+      this.setState(newState);
+
+      if ($.isEmptyObject(this.state.validationErrors) == false) return null;
```

Comment:
This returns null and if the if is false, it will return undefined. Sure you want to do that? 

Also, the $ usage can be replaced by using the lodash library. It doesn't even look like you imported jquery, but please don't.